transcom/mymove
# [MB-16384] Authz check for Moving Expense creation endpoint

## [Jira ticket](https://dp3.atlassian.net/jira/software/c/projects/MB/boards/26?modal=detail&selectedIssue=MB-16384)

## Summary

This PR ensures that POST requests to `/ppm-shipments/{ppmShipmentID}/moving-expenses`, the internal endpoint to create moving expenses on a PPM shipment, returns a `404` error if the shipment's service member does not match that of the logged-in user. 

Note that this closely matches the update to the pro-gear weight ticket endpoint on [this PR](https://github.com/transcom/mymove/pull/11099#issuecomment-1652762936); to match as closely as possible, I updated the moving expenses endpoint to return a `201` (successfully created) rather than a `200`, as it did before, which seems a more appropriate status. The check is also done at the service object level, rather than in the handler.

### How to test

1. Find an approved PPM Shipment, or create one: 
    1. Sign into the customer app and create a move with a PPM shipment
    2. Sign into the office app as a service counselor in the same gbloc
    3. Find and approve the shipment created in step 1
2. Note the ID of the PPM shipment.
3. Authenticate in Postman as a customer other than the one above. Note that the `X-CSRF-TOKEN` header should be set to the value of the `masked_gorilla_csrf` cookie.
4. Send a POST request to `/ppm-shipments/{ppmShipmentId}/moving-expenses`, with the ID noted above as the `ppmShipmentId`.
5. The response should error with a `404`; the response would be a `200`, with the expense created on the PPM, without this change (on `main`).
6. Server tests should pass.
